src/audio/drc/drc.c

```diff
@@ -261,6 +302,10 @@ static int drc_process(struct processing_module *mod,
 		}
 	}
 
+	/* Control pass-though in processing function with switch control */
+	if (drc_has_config(cd))
+		cd->enabled = (drc_get_config_enabled(cd) && cd->enable_switch) ? true : false;
```

Comment:
`&&` is a logical operation, its result is already a `true` or a `false`, so a simple `cd->enabled = drc_get_config_enabled(cd) && cd->enable_switch` isenough